mappings/net/minecraft/world/gen/ChunkRandom.mapping

```diff
@@ -2,32 +2,77 @@ CLASS net/minecraft/class_2919 net/minecraft/world/gen/ChunkRandom
 	FIELD field_13276 sampleCount I
 	METHOD <init> (J)V
 		ARG 1 seed
-	METHOD method_12659 setSeed (II)J
-		ARG 1 x
-		ARG 2 z
+	METHOD method_12659 setTerrainSeed (II)J
+		COMMENT Seeds the randomizer to generate the surface terrain blocks(such as grass, sand, etc)
+		COMMENT and the bedrock patterns.
+		COMMENT
+		COMMENT <p>Note that the terrain seed is NOT world seed dependent and only gets affected by
+		COMMENT chunk coordinates.</p>
+		ARG 1 chunkX
+		ARG 2 chunkZ
 	METHOD method_12660 consume (I)V
+		COMMENT Skips the provided number of calls to the randomizer.
+		COMMENT
+		COMMENT <p>The skips give the effect of "scrambling" the randomizer but the output is still
+		COMMENT linearly dependent. Note that since multiple calls to a linear congruential generator is
+		COMMENT equivalent to another linear congruence, this method could be optimized to combine
+		COMMENT the calls into one.</p>
 		ARG 1 count
-	METHOD method_12661 setSeed (JII)J
+	METHOD method_12661 setPopulationSeed (JII)J
+		COMMENT Seeds the randomizer to create population features such as decorators and animals.
+		COMMENT
+		COMMENT <p>This method takes in the world seed and the negative-most block coordinates of the
+		COMMENT chunk. The coordinate pair provided is equivalent to (chunkX * 16, chunkZ * 16). The
+		COMMENT three values are mixed together through some layers of hashing to produce the
+		COMMENT population seed.</p>
+		COMMENT
+		COMMENT <p>This function has been proved to be reversible through some exploitation of the underlying
+		COMMENT nextLong() weaknesses. It is also important to remember that since setSeed()
+		COMMENT truncates the 16 upper bits of world seed, only the 48 lowest bits affect the population
+		COMMENT seed output.</p>
 		ARG 1 worldSeed
-		ARG 3 x
-		ARG 4 z
-	METHOD method_12662 create (IIJJ)Ljava/util/Random;
-		ARG 0 x
-		ARG 1 z
+		ARG 3 posX
+		ARG 4 posZ
+	METHOD method_12662 getSlimeRandom (IIJJ)Ljava/util/Random;
+		ARG 0 chunkX
+		ARG 1 chunkZ
 		ARG 2 worldSeed
-		ARG 4 localSeed
-	METHOD method_12663 setStructureSeed (JII)J
-		ARG 1 worldSeed
-		ARG 3 x
-		ARG 4 z
-	METHOD method_12664 setFeatureSeed (JII)J
+		ARG 4 scrambler
+	METHOD method_12663 setCarverSeed (JII)J
```

Comment:
Need help for a better name. This method is responsible for:

1) Generating all carvers. (caves, ravines, etc)
2) Computing the start chunk of mineshafts. (doesn't use setRegionSeed for some reason.)
3) Computing the stronghold piece layout(not to be confused with the generation.)
4) All StructureFeature's structure starts. Examples of this include rotation, random structure pieces and templates.

Reply:
See my comment above https://github.com/FabricMC/yarn/pull/1118#discussion_r380219962

Reply:
> Computing the start chunk of mineshafts. (doesn't use setRegionSeed for some reason.)

I think that's just Mojang derping and using the wrong method. Given it's unintentional, I'd say don't take this into account when naming the method.